src/client/mod.rs

```diff
@@ -250,7 +250,11 @@ impl Client {
     }
 
     /// Starts a new `ClientSession`.
-    pub async fn start_session(&self, options: Option<SessionOptions>) -> Result<ClientSession> {
+    pub async fn start_session(
+        &self,
+        options: impl Into<Option<SessionOptions>>,
```

Comment:
I think this is technically a breaking change if someone were relying on the type inference for the options, but I don't really think it's a cause for concern.